pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -407,6 +407,18 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		return reconcile.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// requeue if allowed disruptions in the default PDB is 0
+	allowedDisruptions, err := r.getAllowedDisruptions(osdPDBAppName, request.Namespace)
+	if err != nil {
+		logger.Debugf("failed to get allowed disruptions count from default osd pdb %q. Skipping reconcile. Error: %v", osdPDBAppName, err)
+		return reconcile.Result{}, nil
```

Comment:
Shouldn't we return the error and requeue? It seems like a rare error condition.

Reply:
Only possible error here is that the default PDB (osdPDBAppName) resource is not present. That would happen when blocking PDBs are present instead . Rook would not want to reconcile again in this situation.  So I think its safe to ignore this error.

Reply:
How about if the `getAllowedDisruptions()` method returns the err directly instead of wrapping it, then here we could check for `errors.IsNotFound()`. Then it's obvious that we can safely ignore it if the pdb doesn't exist.